pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -1226,10 +1215,16 @@ func (m *kubeGenericRuntimeManager) SyncPod(ctx context.Context, pod *v1.Pod, po
 	}
 
 	// Step 6: start the init container.
-	if container := podContainerChanges.NextInitContainerToStart; container != nil {
+	for _, idx := range podContainerChanges.InitContainersToStart {
```

Comment:
comment to simplify the review:

there are cases when Sidecar needs to be restarted AND the next Init container has to start. Without the collection we ended up in situaiton when forever-failing sidecar container will block Pod Initialization and will not allow it to work. This is not desireable